subsys/bluetooth/controller/hci_driver.c

```diff
@@ -679,6 +679,13 @@ static int configure_supported_features(void)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_BT_CTLR_SDC_QOS_CHANNEL_SURVEY)) {
+		err = sdc_support_qos_channel_survey();
```

Comment:
Shouldn't be there additional `#ifdefs` to avoid warning during building when feature is not enabled:
```
/home/redbeard/ncs/nrf/subsys/bluetooth/controller/hci_driver.c:684:9: error: implicit declaration of function ‘sdc_support_qos_channel_survey’ [-Werror=implicit-function-declaration]
  684 |   err = sdc_support_qos_channel_survey();
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
?

For example:
```c
#ifdef CONFIG_BT_CTLR_SDC_QOS_CHANNEL_SURVEY
	if (IS_ENABLED(CONFIG_BT_CTLR_SDC_QOS_CHANNEL_SURVEY)) {
		err = sdc_support_qos_channel_survey();
		if (err) {
			return -ENOTSUP;
		}
	}
#endif
```

Reply:
That code is superfluous because the #ifdef and if do the same thing (one is dropped because ifdef is not true, other is dropped because it evaluates to if (0)) so the fix should be that the header be included so that the function definition is always available

Reply:
@nordicjm - yes, you are right. Moreover I discovered, that my problem came from not updated dragoon repository revision:
https://github.com/nrfconnect/sdk-nrf/blob/8633d05d4d445fe1ab329275849c6320a5bac998/west.yml#L172-L179

When I updated revision to current HEAD (and apply west update), then my problem with lack of linking disappear. So, there is no need to add extra code - just dragoon revision in west manifest should be updated.